File: src/com/github/gtache/editor/listeners/EditorMouseListenerImpl.scala
package com.github.gtache.editor.listeners

import com.intellij.openapi.editor.event.{EditorMouseEvent, EditorMouseListener}

/**
  * An EditorMouseListener implementation which just listens to mouseExited and mouseEntered
  */
class EditorMouseListenerImpl extends EditorMouseListener with LSPListener {

  override def mouseExited(e: EditorMouseEvent): Unit = {
    if (checkManager()) manager.stopListening()
  }

  override def mousePressed(e: EditorMouseEvent): Unit = {
  }

  override def mouseReleased(e: EditorMouseEvent): Unit = {
  }

  override def mouseEntered(e: EditorMouseEvent): Unit = {
    if (checkManager()) manager.startListening()
  }

  override def mouseClicked(e: EditorMouseEvent): Unit = {
  }
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/editor/listeners/EditorMouseListenerImpl.kt
package com.github.gtache.lsp.editor.listeners

import com.intellij.openapi.editor.event.EditorMouseEvent
import com.intellij.openapi.editor.event.EditorMouseListener

/**
 * An EditorMouseListener implementation which just listens to mouseExited and mouseEntered
 */
class EditorMouseListenerImpl : EditorMouseListener, AbstractLSPListener() {

    override fun mouseExited(e: EditorMouseEvent): Unit {
        if (checkEnabled()) manager?.mouseExited()
    }

    override fun mousePressed(e: EditorMouseEvent): Unit {
    }

    override fun mouseReleased(e: EditorMouseEvent): Unit {
    }

    override fun mouseEntered(e: EditorMouseEvent): Unit {
        if (checkEnabled()) manager?.mouseEntered()
    }

    override fun mouseClicked(e: EditorMouseEvent): Unit {
        if (checkEnabled()) manager?.mouseClicked(e)
    }
}